### PR TITLE
feat(core): Global Context

### PR DESCRIPTION
**Pull request by jordankzf**
Global SadoConnectProvider context + various bug fixes

### CODE REVIEW

File: packages/sado-connect/package.json [hunk 1]
```diff
@@ -56,7 +56,8 @@
     "@esbuild-plugins/node-globals-polyfill": "^0.2.3",
     "@esbuild-plugins/node-modules-polyfill": "^0.2.2",
     "@headlessui/react": "^1.7.15",
-    "@sadoprotocol/ordit-sdk": "^0.0.4",
-    "boring-avatars": "^1.10.1"
+    "@sadoprotocol/ordit-sdk": "^0.0.7",
+    "boring-avatars": "^1.10.1",
+    "vite-plugin-css-injected-by-js": "^3.1.2"
```
Comment (kevzzsk) [inline packages/sado-connect/package.json:61]: why did you add `"vite-plugin-css-injected-by-js"` as deps and not devdeps

Comment (jordankzf) [reply packages/sado-connect/package.json]: Oops. Fixed.

File: packages/sado-connect/src/main.tsx [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+import "./style.css";
 
 ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
   <React.StrictMode>
-    <AddressProvider>
-      <SadoConnectKit />
-    </AddressProvider>
+    <SadoConnectProvider>
+      <SadoConnectKit
+        onViewWallet={() => console.log("You can provide a callback here.")}
```
Comment (kyleleow) [inline packages/sado-connect/src/main.tsx:11]: instead of using console.log, let's add comment to the component to describe the props

Comment (kyleleow) [reply packages/sado-connect/src/main.tsx:11]: we can use jsdoc style

Comment (jordankzf) [reply packages/sado-connect/src/main.tsx:11]: Can you link me to an example?

Comment (kyleleow) [reply packages/sado-connect/src/main.tsx:11]: https://github.com/BirthdayResearch/jellyfishsdk/blob/main/packages/whale-api-client/src/api/governance.ts#L7

Comment (kyleleow) [reply packages/sado-connect/src/main.tsx:11]: fyi, in vscode we can type `/**` and it will have jsdoc auto complete option

Comment (jordankzf) [reply packages/sado-connect/src/main.tsx]: added jsdoc

File: packages/sado-connect/src/components/SadoConnectKit.tsx [hunk 3]
```diff
@@ -11,6 +11,15 @@ export interface SadoConnectKitProp {
   onViewWallet?: () => void;
 }
 
+/**
+ * Parent React component for SadoConnectKit, in the form of a button.
+ *
+ * @component
+ * @param {string} [props.customLabel="Connect wallet"] - Custom label for the button.
+ * @param {string} [props.customStyle] - Custom CSS style for the button.
+ * @param {Function} [props.onViewWallet] - Callback function to handle viewing wallet.
```
Comment (kyleleow) [inline packages/sado-connect/src/components/SadoConnectKit.tsx:20]: should mention optional too?

File: packages/sado-connect/src/providers/SadoContext.tsx [hunk 4]
```diff
@@ -11,6 +11,28 @@ interface SadoContextI {
 
 const SadoContext = createContext<SadoContextI>(undefined as any);
 
+/**
+ * (Optionally) global context provider for SadoConnectKit and its consumer(s).
```
Comment (kyleleow) [inline packages/sado-connect/src/providers/SadoContext.tsx:15]: do you mean `Optional`?

Comment (jordankzf) [reply packages/sado-connect/src/providers/SadoContext.tsx:15]: (Optionally) global / Global (optional)

The grammar is correct.

File: packages/sado-connect/src/components/SadoConnectKit.tsx (hunk 3 above)
Comment (jordankzf) [reply packages/sado-connect/src/components/SadoConnectKit.tsx:20]: https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

I think it's already correct

Comment (kyleleow) [reply packages/sado-connect/src/components/SadoConnectKit.tsx:21]: i mean in the description

Comment (jordankzf) [reply packages/sado-connect/src/components/SadoConnectKit.tsx:21]: I think that would be redundant

**Review by kyleleow — APPROVED** [review]

**Review by JoshuaC817 — APPROVED** [review]
